### PR TITLE
Clean up for the initial external project access scanner PR

### PR DESCRIPTION
**Pull request by angelsungoogle**
clean up for the initial external project access scanner PR #2004 

Thanks for opening a Pull Request!

Here's a handy checklist to ensure your PR goes smoothly.

- [x] I signed Google's [Contributor License Agreement](https://opensource.google.com/docs/cla/)
- [x] My code conforms to Google's [python style](https://google.github.io/styleguide/pyguide.html).
- [x] My PR at a minimum doesn't decrease unit-test coverage (if applicable).
- [x] My PR has been functionally tested.
- [x] All of the [unit-tests](https://forsetisecurity.org/docs/latest/develop/dev/testing.html) still pass.
- [x] Running `pylint --rcfile=pylintrc` passes.

These guidelines and more can be found in our [contributing guidelines](https://github.com/GoogleCloudPlatform/forseti-security/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests/scanner/resource_util_test.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+from google.cloud.forseti.services.inventory.storage import Storage
+from tests.common.gcp_api.test_data import fake_crm_responses
+from tests.common.gcp_api.test_data import http_mocks
+from tests.services.api_tests.inventory_test import TestServiceConfig
+from tests.services.inventory.storage_test import ResourceMock
+from tests.services.util.db import create_test_engine
+from tests.unittest_utils import ForsetiTestCase
+from tests.unittest_utils import get_datafile_path
+
+class OtherModuleTest(ForsetiTestCase):
```
Comment (blueandgold) [inline tests/scanner/resource_util_test.py]: This class name should be `ResourceUtilTest`

File: tests/scanner/resource_util_test.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            credentials=mock_creds)
+        self.project_id = fake_crm_responses.FAKE_PROJECT_ID
+
+    def test_cast_to_gcp_resources(self):
+        """Validate get_project_ancestry_resources() with test project."""
+        expected_cast_resources = "[project<id=forseti-system-test,parent=None>, folder<id=3333333,parent=None>, organization<id=2222222,parent=None>]"
+        api_result = [{u'resourceId': {u'type': u'project', u'id': u'forseti-system-test'}}, {u'resourceId': {u'type': u'folder', u'id': u'3333333'}}, {u'resourceId': {u'type': u'organization', u'id': u'2222222'}}]
+
+        cast_resources = resource_util.cast_to_gcp_resources(api_result)
+        self.assertEqual(expected_cast_resources, str(cast_resources))
```
Comment (blueandgold) [inline tests/scanner/resource_util_test.py]: add a newline at the end of this file

File: tests/scanner/scanners/external_project_access_scanner_test.py [hunk 3]
```diff
@@ -239,51 +240,9 @@ def test_get_emails(self, mock_get_latest_inv_ndx_id, mock_storage):
...
-
-        cast_resources = resource_util.cast_to_gcp_resources(result)
-
-        self.assertTrue(isinstance(cast_resources, list))
 
-        self.assertTrue(isinstance(cast_resources[0], Project))
-        self.assertTrue(isinstance(cast_resources[-1], Organization))
 
-def reduced_inventory(storage, types):
+def get_inventory(storage, types):
```
Comment (blueandgold) [inline tests/scanner/scanners/external_project_access_scanner_test.py]: Is this `reduced_inventory()` being called from somewhere?  Do we need to change the caller?

File: tests/scanner/audit/external_project_access_engine_test.py [hunk 4]
```diff
@@ -135,7 +135,7 @@ class ExternalProjectAccessRuleBookTest(ForsetiTestCase):
     TEST_ANCESTORS = [Project('123'),
                       Folder('456'),
                       Organization('7890')]
-    TEST_BAD_ANC = [Project('123'),
+    TEST_BAD_ANCESTORS = [Project('123'),
                     Folder('456'),
```
[the remark this answers is not in the document]
Comment (angelsungoogle) [reply tests/scanner/audit/external_project_access_engine_test.py]: revised as suggested

File: tests/scanner/resource_util_test.py (hunk 1 above)
Comment (angelsungoogle) [reply tests/scanner/resource_util_test.py]: modified as suggested

File: tests/scanner/resource_util_test.py (hunk 2 above)
Comment (angelsungoogle) [reply tests/scanner/resource_util_test.py]: modified as suggested

File: tests/scanner/scanners/external_project_access_scanner_test.py (hunk 3 above)
Comment (angelsungoogle) [reply tests/scanner/scanners/external_project_access_scanner_test.py]: deleted

File: tests/scanner/resource_util_test.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+from datetime import datetime
+import json
+# pylint says unittest goes before mock
+import unittest
+import mock
+
+# pylint says sqlalchemy.orm goes before google.auth
```
Comment (blueandgold) [inline tests/scanner/resource_util_test.py]: remove this comment

File: tests/scanner/resource_util_test.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+
+from google.cloud.forseti.common.gcp_type import resource_util
+from tests.unittest_utils import ForsetiTestCase
+
+class ResourceUtilTest(ForsetiTestCase):
+    """Test misc."""
```
Comment (blueandgold) [inline tests/scanner/resource_util_test.py]: Update this docstring:
```
"""Test for the Resource Util."""
```

**Pull request merged**